numpy/core/src/umath/loops_trigonometric.dispatch.c.src

```diff
@@ -66,6 +66,98 @@ simd_@op@_scalar_f64(npyv_f64 out, npy_uint64 cmp_bits)
 }
 /**end repeat**/
 
+#if NPY_SIMD_CVT_F64
+#if defined(NPY_OS_WIN32) || defined(NPY_OS_CYGWIN)
+NPY_FINLINE npyv_f64
+#else
+NPY_NOINLINE npyv_f64
+#endif
+simd_tan_scalar_f64(npyv_f64 out)
+{
+    npy_double NPY_DECL_ALIGNED(NPY_SIMD_WIDTH) out_copy[npyv_nlanes_f64];
+    npyv_storea_f64(out_copy, out);
+
+    for (unsigned i = 0; i < npyv_nlanes_f64; ++i) {
+        out_copy[i] = npy_tan(out_copy[i]);
+    }
+
+    return npyv_loada_f64(out_copy);
+}
+
+/* Vector approximation for double-precision tan.
+   Maximum measured error is 3.48 ULP:
+   __v_tan(0x1.4457047ef78d8p+20) got -0x1.f6ccd8ecf7dedp+37
+				 want -0x1.f6ccd8ecf7deap+37.   */
+NPY_FINLINE
+npyv_f64 simd_inner_tan(npyv_f64 x) {
+    const npyv_f64 shift = npyv_setall_f64(0x1.8p52);
+    const npyv_u64 abs_mask = npyv_setall_u64(0x7fffffffffffffff);
+    const npyv_u64 range_val = npyv_setall_u64(0x4160000000000000);
+    const npyv_u64 tiny_bound = npyv_setall_u64(0x3e50000000000000);
+    const npyv_f64 two_over_pi = npyv_setall_f64(0x1.45f306dc9c883p-1);
+    const npyv_f64 half_pi_high = npyv_setall_f64(-0x1.921fb54442d18p0);
+    const npyv_f64 half_pi_low = npyv_setall_f64(-0x1.1a62633145c07p-54);
+    const double poly[9] = {
+        0x1.5555555555556p-2, 0x1.1111111110a63p-3, 0x1.ba1ba1bb46414p-5,
+        0x1.664f47e5b5445p-6, 0x1.226e5e5ecdfa3p-7, 0x1.d6c7ddbf87047p-9,
+        0x1.7ea75d05b583ep-10, 0x1.289f22964a03cp-11, 0x1.4e4fd14147622p-12
+    };
+
+    npyv_u64 iax = npyv_and_u64(npyv_reinterpret_u64_f64(x), abs_mask);
+
+    /* Our argument reduction cannot calculate q with sufficient accuracy for very
+        large inputs. Fall back to scalar routine for all lanes if any are too
+        large, or Inf/NaN. If fenv exceptions are expected, also fall back for tiny
+        input to avoid underflow. Note pl does not supply a scalar double-precision
+        tan, so the fallback will be statically linked from the system libm.  */
+    npyv_b64 check = npyv_cmpgt_u64(npyv_sub_u64(iax, tiny_bound), npyv_sub_u64(range_val, tiny_bound));
+    if (npyv_any_b64(check)) {
+        return simd_tan_scalar_f64(x);
+    }
+
+    /* q = nearest integer to 2 * x / pi.  */
+    npyv_f64 q = npyv_sub_f64(npyv_muladd_f64(x, two_over_pi, shift), shift);
+    npyv_s64 qi = npyv_cvt_s64_f64(q);
+
+    /* Use q to reduce x to r in [-pi/4, pi/4], by:
+        r = x - q * pi/2, in extended precision.  */
+    npyv_f64 r = x;
+    r = npyv_muladd_f64(q, half_pi_high, r);
```

Comment:
```suggestion
    r = npyv_nmuladd_f64(q, half_pi_high, r);
```
note that you could still use `vfmsq_f64` with no need to modify the sign of the original constant.